python/cli/data_platform_workflows_cli/update_bundle.py

```diff
@@ -66,10 +74,22 @@ def main():
     # Full list of possible series config (unsupported) can be found under "Charm series" at https://juju.is/docs/olm/bundle
     default_series = file_data.get("series")
     for app in file_data["applications"].values():
-        app["revision"] = fetch_latest_revision(
-            app["charm"], app["channel"], app.get("series", default_series)
-        )
+        channel_map, resources = fetch_charm_info_from_store(app['charm'], app['channel'])
 
+        if latest_revision := fetch_latest_revision(channel_map, app.get("series", default_series)):
+            app["revision"] = latest_revision
+        else:
+            raise ValueError(
+                f"Revision not found for {app['charm']} on {app['channel']} for Ubuntu {app.get('series', default_series)}"
+            )
+        for resource in resources:
+            if resource["type"] == "oci-image":
+                image_path, src_creds = fetch_oci_image_metadata(resource["download"]["url"])
+                app["resources"] = {
+                    resource["name"]: resource["revision"],
+                    "oci-image": image_path,
+                    "src-creds": src_creds,
```

Comment:
included the credentials because they are necessary for the `skopeo` set when setting up the airgapped env, and they are pretty much public anyway (public snapcraft api returns it without auth) so my idea was to add it in order for the bundle to contain all information one could need when deploying on airgapped. LMK if you don't want it and I remove it!